docs/csharp/programming-guide/main-and-command-args/top-level-statements.md

```diff
@@ -0,0 +1,90 @@
+
+A project can have any number of additional source code files that don't have top-level statements.
+
+## No other entry points
+
+You can write a `Main` method explicitly, but it can't function as an entry point. The compiler issues the following warning:
+
+> CS7022 The entry point of the program is global code; ignoring 'Main()' entry point.
+
+In a project with top-level statements, you can't use the [-main](../../language-reference/compiler-options/main-compiler-option.md) compiler option to select the entry point, even if the project has one or more `Main` methods.
+
+## `using` directives
+
+If you include using directives, they must come first in the file, as in this example:
+
+:::code language="csharp" source="snippets/top-level-statements-1/Program.cs":::
+
+## Global namespace
+
+Top-level statements are implicitly in the global namespace.
+
+## Namespaces and type definitions
+
+A file with top-level statements can also contain namespaces and type definitions, but they must come after the top-level statements. For example:
+
+:::code language="csharp" source="snippets/top-level-statements-2/Program.cs":::
+
+## `args`
+
+Top-level statements can reference the `args` variable to access any command-line arguments that were entered. The `args` variable is never null but its `Length` is zero if no command-line arguments were provided. For example:
+
+:::code language="csharp" source="snippets/top-level-statements-3/Program.cs":::
+
+## `await`
+
+You can call an async method by using `await`. For example:
+
+:::code language="csharp" source="snippets/top-level-statements-4/Program.cs":::
+
+## Exit code for the process
+
+To return an `int` value when the application ends, use the `return` statement as you would in a `Main` method that returns an `int`. For example:
+
+:::code language="csharp" source="snippets/top-level-statements-5/Program.cs":::
+
+## Implicit entry point method
+
+The compiler generates a method to serve as the program entry point for a project with top-level statements. The name of this method isn't actually `Main`, it's an implementation detail that your code can't reference directly. The signature of the method depends on whether the top-level statements contain the `await` keyword or the `return` statement. The following table shows what the method signature would look like, using the method name `Main` in the table for convenience.
+
+| Top-level code contains| Implicit `Main` signature                    |
+|------------------------|----------------------------------------------|
+| `await` and `return`   | `static async Task<int> Main(string[] args)` |
+| `await`                | `static async Task Main(string[] args)`      |
+| `return`               | `static int Main(string[] args)`             |
+| No `await` or `return` | `static void Main(string[] args)`            |
+
+## C# language specification
+
+[!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]
+
```

Comment:
I'd add a link to the [top level statements speclet](https://docs.microsoft.com/dotnet/csharp/language-reference/proposals/csharp-9.0/top-level-statements). The link would be `~/_csharplang/proposals/csharp-9.0/top-level-statements.md`